bin/workspace_root.mli

```diff
@@ -16,4 +16,5 @@ type t =
   ; kind : Kind.t
   }
 
-val create : default_is_cwd:bool -> specified_by_user:string option -> t
+val create : default_is_cwd:bool -> specified_by_user:string option -> (t, exn) result
```

Comment:
How about just returning `User_error.t`? Usually it's a mistake to delay raising other types of exceptions (code errors for example)

Reply:
Good point, fixed.